easy-rules-jexl/src/test/java/org/jeasy/rules/jexl/JexlRuleFactoryTest.java

```diff
@@ -0,0 +1,191 @@
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package org.jeasy.rules.jexl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.Reader;
+import java.io.StringReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+
+import org.jeasy.rules.api.Rule;
+import org.jeasy.rules.api.Rules;
+import org.jeasy.rules.support.JsonRuleDefinitionReader;
+import org.jeasy.rules.support.UnitRuleGroup;
+import org.jeasy.rules.support.YamlRuleDefinitionReader;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class JexlRuleFactoryTest {
+
+    @Parameters
+    public static Collection<Object[]> params() {
+        return Arrays.asList(new Object[][] {
+                { new JexlRuleFactory(new YamlRuleDefinitionReader()), "yml" },
+                { new JexlRuleFactory(new JsonRuleDefinitionReader()), "json" },
+        });
+    }
+
+    @SuppressWarnings("deprecation")
+    @org.junit.Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Parameter(0)
+    public JexlRuleFactory factory;
+
+    @Parameter(1)
+    public String ext;
+
+    @Test
+    public void testRulesCreation() throws Exception {
+        // given
+        File rulesDescriptor = new File("src/test/resources/rules." + ext);
+
+        // when
+        Rules rules = factory.createRules(new FileReader(rulesDescriptor));
+
+        // then
+        assertThat(rules).hasSize(2);
+        Iterator<Rule> iterator = rules.iterator();
+
+        Rule rule = iterator.next();
+        assertThat(rule).isNotNull();
+        assertThat(rule.getName()).isEqualTo("adult rule");
+        assertThat(rule.getDescription()).isEqualTo("when age is greater then 18, then mark as adult");
+        assertThat(rule.getPriority()).isEqualTo(1);
+
+        rule = iterator.next();
+        assertThat(rule).isNotNull();
+        assertThat(rule.getName()).isEqualTo("weather rule");
+        assertThat(rule.getDescription()).isEqualTo("when it rains, then take an umbrella");
+        assertThat(rule.getPriority()).isEqualTo(2);
+    }
+
+    @Test
+    public void testRuleCreationFromFileReader() throws Exception {
+        // given
+        Reader adultRuleDescriptorAsReader = new FileReader("src/test/resources/adult-rule." + ext);
+
+        // when
+        Rule adultRule = factory.createRule(adultRuleDescriptorAsReader);
+
+        // then
+        assertThat(adultRule.getName()).isEqualTo("adult rule");
+        assertThat(adultRule.getDescription()).isEqualTo("when age is greater then 18, then mark as adult");
```

Comment:
There is a typo here: `then` should be `than`. It was fixed in 94d29cc86b716f9be6de8a07c411940db25018f7. Please fix it here as well. (there are a couple of other occurrences in this PR).